symfony/console/7.0/bin/console

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env php
+<?php
+
+use App\Kernel;
+use Symfony\Bundle\FrameworkBundle\Console\Application;
+
+if (!is_file(dirname(__DIR__).'/vendor/autoload_runtime.php')) {
+    throw new LogicException('Symfony Runtime is missing. Try running "composer require symfony/runtime".');
+}
+
+require_once dirname(__DIR__).'/vendor/autoload_runtime.php';
+
+return function (array $context) {
+    $kernel = new Kernel($context['APP_ENV'], (bool) $context['APP_DEBUG']);
+
+    $app = new Application($kernel);
+    $app->setCatchErrors('prod' === $context['APP_ENV']);
```

Comment:
Why would you want to ever not catch errors? :thinking:

Reply:
I see it the same way as you. But I tried to keep it in the _spirit_ of comments at this [pull-request](https://github.com/symfony/symfony/pull/50420)

While developing or testing, you might not want it to happen? But in `prod` definitely catch it.

Reply:
Sorry, which comment(s) are you referring to?

Reply:
https://github.com/symfony/symfony/pull/50420#issuecomment-1576688896, https://github.com/symfony/symfony/pull/50420#issuecomment-1576712904 and https://github.com/symfony/symfony/pull/50420#issuecomment-1576849725

> What about adding a flag to choose if we should handle throwable or not? 

> Sounds like a good idea 👍

> ...
> And if the developer wants more output, they can run the same command with -v flags

Maybe I am misunderstanding, but it was my impression that the ability to catch was intended to be optional.

Reply:
The ability to catch is optional/configurable, but:

- given how bad Symfony behaves when it's off
- given the default value for errors is inconsistent with the default value for exceptions
- given inconsistencies between dev and prod might lead to WTF

I'd recommend to aim for something simple: the same experience everywhere.